examples/decodeFrame.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+import canmatrix.formats
+import sys
+
+# command line options...
+usage = """
+%prog [options]  matrix  frame
+
+matrixX can be any of *.dbc|*.dbf|*.kcd|*.arxml
+frame is AAA#YYYYYYYYYYYYYYYY or
+       BBBBB#YYYYYYYYYYYYYYYY or
+
+
+where AAA is standard ID and BBBBB is extended ID
+
+"""
+
+if len(sys.argv) < 3:
+    print(usage)
+    sys.exit(1)
+
+# load matrix
+db = canmatrix.formats.loadp(sys.argv[1], flat_import=True)
+
+# load frame data from argv
+frame_string = sys.argv[2]
+(arbitration_id_string, hexdata) = frame_string.split('#')
+
+# set arbitration_id
+arbitration_id = canmatrix.ArbitrationId(int(arbitration_id_string, 16))
+arbitration_id.id = int(arbitration_id_string, 16)
+
+# extended frame?
+if len(arbitration_id_string) == 3:
```

Comment:
not really, the `from_compound_integer` does another thing:
 `from_compound_integer` expects a special bit (Bit 31) to be set and tell me, the arbitration id is extended.
Here I get the information if it's extended id from string length of the given id (compare to candump).

Reply:
Ok I see. Than there is risk that you want to decode can ID 0x0DF and writes only `DF#0123FFAAFF` instead of `0DF#0123FFAAFF` and it will be evaluated as the extended id. Maybe `if len(arbitration_id_string) <= 3` would be more safe?

Reply:
I agree